pkgs/stdenv/generic/check-meta.nix

```diff
@@ -733,6 +736,12 @@ let
               # 2) locally set through API
               if purlPartsFormatted != null then
                 [ purlPartsFormatted ]
+              else if
+                !evaluateSrc
+                || !(builtins.tryEval attrs.src or [ ]).success
+                || !(builtins.tryEval attrs.srcs or [ ]).success
```

Comment:
That's precisely the stuff I am not happy about, because we can't make a meaningful difference between asserts/throws that are "good to ignore" and those that actually are a problem. We use both across the tree.